python-backend/app/api/search/search/resources/search.py

```diff
@@ -0,0 +1,132 @@
+                grouped_results[result.result['id']].score += result.score
+            else:
+                grouped_results[result.result['id']] = result
+
+        top_search_results = list(grouped_results.values())
+        top_search_results.sort(key=lambda x: x.score, reverse=True)
+
+        all_search_results = {}
+
+        for type in search_types:
+            top_search_results_by_type = {}
+
+            max_results = 5
+            if len(search_types) == 1:
+                max_results = 50
+
+            for result in top_search_results:
+                if len(top_search_results_by_type) > max_results:
+                    break
+                if result.type == type:
+                    top_search_results_by_type[result.result['id']] = result
+
+            full_results = db.session.query(search_targets[type]['model']).filter(
+                search_targets[type]['primary_column'].in_(
+                    top_search_results_by_type.keys())).all()
+
+            for full_result in full_results:
+                top_search_results_by_type[getattr(
+                    full_result, search_targets[type]['id_field'])].result = full_result
+
+            all_search_results[type] = list(top_search_results_by_type.values())
+
+        return {'search_terms': search_terms, 'search_results': all_search_results}
+
+
+def append_result(search_results, search_term, type, item, id_field, value_field, score_multiplier):
+
+    if getattr(item, value_field).lower().startswith(search_term.lower()):
+        score_multiplier = score_multiplier * 3
+
+    if getattr(item, value_field).lower() == search_term.lower():
+        score_multiplier = score_multiplier * 10
+
+    search_results.append(
+        SearchResult(
+            getattr(item, 'score') * score_multiplier, type, {
+                'id': getattr(item, id_field),
+                'value': getattr(item, value_field)
+            }))
+
+
+def execute_search(app, search_results, search_term, search_terms, type, type_config):
+    with app.app_context():
+        for term in search_terms:
+            if len(term) > 2:
+                for column in type_config['columns']:
+                    similarity = db.session.query(Mine).with_entities(
+                        func.similarity(column, term).label('score'),
+                        *type_config['entities']).filter(column.ilike(f'%{term}%'))
+                    if type_config['has_deleted_ind']:
+                        similarity = similarity.filter_by(deleted_ind=False)
+                    similarity = similarity.order_by(desc(func.similarity(column, term))).all()
+                    [
+                        append_result(search_results, search_term, type, item,
+                                      type_config['id_field'], type_config['value_field'],
+                                      type_config['score_multiplier']) for item in similarity
+                    ]
```

Comment:
weird usage of list comprehension syntax, just loop with `for item in similarity`?

Reply:
Changed